python/taichi/lang/error.py

```diff
@@ -0,0 +1,59 @@
+import functools
+import traceback
+from colorama import Fore, Style
+import sys
+
+
+def no_traceback(foo):
+    '''
+    Don't include the decorated frame in traceback, if ti.set_excepthook() was called.
+    '''
+    @functools.wraps(foo)
+    def wrapped(*args, **kwargs):
+        _taichi_skip_traceback = 2
+        return foo(*args, **kwargs)
+
+    return wrapped
+
+
+def enable_excepthook():
+    def excepthook(exctype, value, tb):
+        skip = 0
+        back = 4
+        forward = 2
+        bar = f'{Fore.LIGHTBLACK_EX}{"-"*44}{Fore.RESET}'
+        print(
+            f'{Fore.LIGHTBLACK_EX}========== Taichi Stack Traceback =========={Fore.RESET}'
+        )
+        for frame, lineno in traceback.walk_tb(tb):
+            name = frame.f_code.co_name
+            filename = frame.f_code.co_filename
+            if '_taichi_skip_traceback' in frame.f_locals:
+                skip = frame.f_locals['_taichi_skip_traceback']
```

Comment:
I wonder if this should be `+=` or `=`? What if `skip` is already positive?

Reply:
It will be overriden..

This enable such usage:
```
def func1():
  _taichi_skip_traceback = 233  # I want to disable all traceback below me!
  func2()

def func2():
  _taichi_skip_traceback = 0  # except, me! I'll always be visiable from traceback no matter where I'm called from
```

Reply:
I see...